stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/kv/RocksdbKVStore.java

```diff
@@ -367,11 +376,35 @@ protected void openRocksdb(StateStoreSpec spec) throws StateStoreException {
             }
 
             List<ColumnFamilyHandle> cfHandles = Lists.newArrayListWithExpectedSize(2);
-            RocksDB db = RocksDB.open(
-                options,
-                dbDir.getAbsolutePath(),
-                Lists.newArrayList(metaDesc, dataDesc),
-                cfHandles);
+
+            RocksDB db = TtlDB.open(
```

Comment:
if ttlSeconds is not set, can we simply open the Database in the old way ?

every existing BK cluster does not have this option
IIUC with this change we are going to open/close/open the DB
we can go with `RocksDB.open` without even trying to do `TtlDB.open`

Reply:
Makes sense. I was hoping to eventually remove the conditional logic, but with no practical way to upgrade existing DBs…

I’ll simplify the logic and change the error message to indicate what is wrong.  This will also enable a BK rollback if the option hasn’t been used yet to create any new tables with a ttl.

I need to think about a way to fail opening of a ttl db with the old code. Using different column family names should suffice.

Reply:
Updated to check that an existing DB is being opened with a TTL settings compatible with a previous open.  This provides a better error indication upon a mismatch.  This also avoids re-playing a writeahead log against the wrong implementation; this isn't a current concern but a subtle bug if someone changes the code to use a local WAL.

Reply:
FYI @eolivelli